cpp/Enums.cpp

```diff
@@ -71,6 +72,7 @@ namespace
 		static_assert(Repetition::REQUIRED == 0);
 		static_assert(Repetition::OPTIONAL == 1);
 		static_assert(Repetition::REPEATED == 2);
+		static_assert(Repetition::UNDEFINED == 3);
```

Comment:
Whilst we're at it, we should add `Encoding::UNKNOWN` with value `999`.
See https://github.com/apache/arrow/blob/apache-arrow-5.0.0/cpp/src/parquet/types.h#L478

Reply:
I've left it out since we currently do not expose `Encoding.Unknown` in C#. Thus no risk of ABI breaks here.

Also, a bit of a WTF here. What's the difference from Arrow's point of view between `Encoding::UNDEFINED` and `Encoding::UNKNOWN` ?

Reply:
Thanks, I get your point about ABI breaks. Why do we start exposing `Repetition.Undefined` in C# now though?